src/dataset.rs

```diff
@@ -400,6 +400,13 @@ impl Dataset {
         Ok(Dataset { c_dataset })
     }
 
+    /// Flush all write cached data to disk.
+    ///
+    /// See [`GDALFlushCache`].
+    pub fn flush_cache(&self) {
+        unsafe { GDALFlushCache(self.c_dataset) }
+    }
+
```

Comment:
This is technically not necessary in this PR. I used it to help debug a Linux testing issue and decided to leave it in. Can remove if someone has a problem exposing this.

Reply:
No, I think we should have it, but should it take `&self` or `&mut self`. I guess it's mostly academic with `Dataset` being `!Sync`, though.

Reply:
@lnicola This is similar to something I still find really confusing in this library (or maybe Rust?): you can have a non-mutable reference to a `Dataset`, yet get a `RasterBand` from it and change, say, the `no_data` value. I don't think I have the correct mental model for `& mut` yet.